File: lab 5/insertion.py
import time
import random
def Insertion(A,n):
    for i in range(1,n):
      j=i-1
      curr=A[i]
      while(A[j]>curr and j>=0):
         A[j+1]=A[j]
         j-=1
      A[j+1]=curr

n=int(input("Enter n "))
A=[random.randint(0,1000) for _ in range(n)]
print(A)
t1=time.time()
Insertion(A,n)
t2=time.time()
print(A)
print(f"Time taken is {t2-t1}")

File: lab 5/insertion.c
#include<stdio.h>
#include<stdlib.h>
#include<time.h>
void Insertion(int A[],int n){
    int i,j,curr,prev;
    for(i=1;i<n;i++){
        prev=i-1;
        curr=A[i];
        while(prev>=0 && A[prev]>curr )
        {
            A[prev+1]=A[prev];
            prev--;
        }
        A[prev+1]=curr;
        
    }
    
}
void display(int A[], int n){
    for(int i=0;i<n;i++){
        printf("%d ",A[i]);
    }
    printf("\n");
}
int main()
{
    int i,n;
    time_t start, end;
    int A[200000];
    printf("Enter n ");
    scanf("%d",&n);
    for(i=0;i<n;i++)
        A[i] = rand();
    display(A,n);
    start = time(NULL);
    Insertion(A,n);
    end = time(NULL);
    display(A,n);
    printf("The time taken is %.5f\n",difftime(end,start));
    return 0;
}